contracts/VaultManager/VaultManager.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./IVaultManager.sol";
+import "./VaultManagerEvents.sol";
+import "../Vault/Vault.sol";
+import "poolz-helper-v2/contracts/GovManager.sol";
+
+contract VaultManager is IVaultManager, VaultManagerEvents,GovManager{
+    mapping(uint => address) public VaultIdToVault;
+    mapping(address => uint) public TokenToVaultId;
+    uint public override TotalVaults;
+
+    mapping(address => bool) public isPermitted;
+
+    modifier vaultExists(uint _vaultId){
+        require(VaultIdToVault[_vaultId] != address(0), "VaultManager: Vault not found");
+        _;
+    }
+
+    function setPermitted(address _address, bool _value) external onlyOwnerOrGov{
+        isPermitted[_address] = _value;
+    }
+
+    function CreateNewVault(address _tokenAddress) external override onlyOwnerOrGov returns(uint vaultId){
+        Vault newVault = new Vault(_tokenAddress);
+        vaultId = TotalVaults;
+        VaultIdToVault[vaultId] = address(newVault);
+        TokenToVaultId[_tokenAddress] = vaultId;
+        TotalVaults++;
+        emit NewVaultCreated(vaultId, _tokenAddress);
+    }
+
+    function DepositeByToken(address _tokenAddress, address from, uint _amount)
+        external
+        override
+        vaultExists(TokenToVaultId[_tokenAddress])
+        returns(uint vaultId)
+    {
+        require(isPermitted[msg.sender], "VaultManager: Not permitted");
+        vaultId = TokenToVaultId[_tokenAddress];
+        Vault(VaultIdToVault[vaultId]).deposit(from, _amount);
```

Comment:
I think need to do a transfer directly. so need to approve only the factory and not each vault